components/Heading/Heading.js

```diff
@@ -1,33 +1,188 @@
-import React from 'react';
+import React, { Component } from 'react';
 import { string, number, oneOfType, bool, node, oneOf } from 'prop-types';
 import classNames from 'classnames';
+import LinkIcon from 'static/images/icons/FontAwesome/link-solid.svg';
 import styles from './Heading.css';
 
-Heading.propTypes = {
-  className: string,
-  id: oneOfType([string, number]), // reference for scroll anchors
-  children: node.isRequired,
-  hasHeadingLines: bool,
-  theme: oneOf(['gray', 'secondary', 'white']),
-};
-
-Heading.defaultProps = {
-  className: undefined,
-  id: '',
-  hasHeadingLines: true,
-  theme: 'gray',
-};
-
-function Heading({ children, className, hasHeadingLines, id, theme }) {
-  const classes = classNames(className, styles.Heading, styles[theme], {
-    [`${styles.headingLines}`]: hasHeadingLines,
-  });
-
-  return (
-    <h2 className={classes} id={id}>
-      {children}
-    </h2>
-  );
+class Heading extends Component {
+  static propTypes = {
+    className: string,
+    id: oneOfType([string, number]), // reference for scroll anchors
+    anchorId: string,
+    customAnchorClass: string,
+    children: node.isRequired,
```

Comment:
If you don't make this children and just make it a prop for a string to render (like `text`), you can dynamically generate the `anchorId`! Two birds with one prop.

Reply:
Nice! Thanks for pointing that out.